### ISSUE TITLE
Classpath Exception in OpenJDK and Subclassing

### ISSUE THREAD

**Issue by AlxSr** [adoptium/adoptium-support#999]
## Question

Hello,

My issue is dedicated not to the installation, but rather using the JDK binaries. Please feel free to address me to the correct support channel in case this is not a suitable one in your opinion.

I have a question about extending and implementing openjdk classes and interfaces.
As far as I understood (see the last paragraph of OpenJDK license), the "CLASSPATH" EXCEPTION TO THE GPL allows one to link to certain classes of the library and to copy and distribute the resulting executable under terms of ones choice.

However, I cannot find anywhere does the term "link" used in the exception cover also subclassing and interface implementation?
In other words, is one allowed to implement, for example, the Callable<T> interface or extend the Thread class and copy and distribute the resulting executable under terms of ones choice (ofc, complying also with other requirements of the last paragraph of the openjdk license file)?

Thank you in advance for the answer.

## Context

Java version:

Not relevant

Your operating system and platform:

Not relevant
